nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/ui/web/auth/service/LoginProviderLink.java

```diff
@@ -93,7 +106,7 @@ public void setIconPath(String iconPath) {
     public String getLink(HttpServletRequest req, String requestedUrl) {
         if (urlComputerClass != null && urlComputer == null) {
             try {
-                urlComputer = (LoginProviderLinkComputer) urlComputerClass.newInstance();
+                urlComputer = urlComputerClass.newInstance();
```

Comment:
`newInstance` directly on class is deprecated in java 9, can you add a `getDeclaredConstructor` ?
```suggestion
                urlComputer = urlComputerClass.getDeclaredConstructor().newInstance();
```